### ISSUE TITLE
Migrate CSS to LESS

### ISSUE THREAD

**Issue by tonypartridge** [JEvents/JEvents#283]
We need to move our CSS to LESS.

This will make our CSS more accurate, cause less template conflicts and remove duplicate code whilst making our life easier to manage.

For now, we will need to assign a watcher in our IDE's to auto generate the CSS files on edit of the LESS files. Rather simple to do. 

I started with the FlatPlus layout already.

**Comment by frogydiak** [comment]
Hi Tony,
Jackson here, I can help on this. I actually started to modify the CSS in the modified version of RSVPro that we're using and the JEvents Layout I developed. In my dev the Layout LESS has this structure:

<Layout Name>
  - assets
     - css
          - less
               - category.less
               - day.less
               - eventdetails.less
               - layout.less
               - location.less
               - month.less
               - search.less
               - tags.less
               - variables.less
               - weeks.less
               - year.less
     - jevents.css
          - jevents.min.css

I broke down the less files into views so it's easier to pinpoint which layout I am trying to work on.

Let me know if you need help on this.

Thanks,

Jackson

**Comment by xtech86** [comment]
Hi Jackson,

Some great feedback! thank you :-)

My concern about this was is duplicating code. For example, it's quite frequent we have reused CSS throughout it without a specific view? Have you checked for duplicate code?

Are the files actually converted to LESS? Or just modified CSS?

Thanks!
Tony

**Comment by frogydiak** [comment]
Hello Tony,
Well, to the best of my skill :) with LESS I guess I have quite use how it should be used but I don't usually go crazy with trying to make it like OOP. 

The main LESS file I called jevents.less instead of using Geraints default events_css.less. The jevents.less file contains just the imports of the other less files so when I edit and save one less file it get compiled. In every view less, I started with prefix that looks something like:

div#jevents {

   // view name
   &.jev-month-view {
      // css for the view here...
  }
}

And all other CSS declaration I added a LESS file called layout.less. This file contains all css that are not for view pages. I also has one for edit.less since I overrided that view.

Is that what you mean with actually converted to LESS? 

Thanks,

Jackson
